### PR TITLE
18GA: Implement Ocilla Southern RR company

### PR DESCRIPTION
**Pull request by perwestling**
Buying corporation receives a free non-sellable 2 train if it has
room for it and the 2 trains hasn't rusted.

### CODE REVIEW

File: lib/engine/step/g_18_ga/buy_company.rb [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+      class BuyCompany < BuyCompany
+        def process_buy_company(action)
+          super
+
+          return unless action.company.sym == 'OSR'
+
+          owner = action.company.owner
+          return if owner.player? || owner.trains.size == @game.phase.train_limit || @game.phase.available?('4')
+
+          free_two_train = @game.trains.select { |t| t.name == '2' }.last.dup
```
Comment (tobymao) [inline lib/engine/step/g_18_ga/buy_company.rb:17]: this should be trains.find  { |t| t.name == '2' }.dup, but what happens if this train is owned by someone else already?

Comment (tobymao) [reply lib/engine/step/g_18_ga/buy_company.rb:17]: why don't you just make a new train?

File: lib/engine/train.rb [hunk 2]
```diff
@@ -10,7 +10,7 @@ class Train
     attr_accessor :obsolete, :operated, :events
     attr_reader :available_on, :name, :distance, :discount, :obsolete_on,
                 :rusts_on, :rusted, :sym, :variant, :variants
-    attr_writer :buyable
+    attr_writer :buyable, :index
```
Comment (tobymao) [inline lib/engine/train.rb:13]: yea i think you should get rid of this and just create the train in buy_company

File: lib/engine/step/g_18_ga/buy_company.rb (hunk 1 above)
Comment (perwestling) [reply lib/engine/step/g_18_ga/buy_company.rb]: Ok, making a new train

File: lib/engine/train.rb (hunk 2 above)
Comment (perwestling) [reply lib/engine/train.rb]: ok

**Pull request merged**